### PR TITLE
Add sub to Debug Counter

### PR DESCRIPTION
**Pull request by zzmp**

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: cf9d160602e35c1dcc37ddf4b47cf8ba133b8905 available at: https://gist.github.com/d63f631183ddd1266082b27213a8d772<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7685.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7685.dmg

File: libraries/shared/src/Debug.h [hunk 1]
```diff
@@ -123,20 +126,25 @@ class Tracker {
...
     } else {
         // Entry for k; update it
-        it->second++;
+        it->second += inc;
     }
 }
 
+template<class K>
+void Counter<K>::sub(const K& k, size_t dec) {
+    add(k, -dec);
```
Comment (Atlante45) [inline libraries/shared/src/Debug.h]: Isn't size_type unsigned?

**Comment by hifi-gustavo** [comment]
Build console output for commit: cf9d160602e35c1dcc37ddf4b47cf8ba133b8905 available at: https://gist.github.com/d63f631183ddd1266082b27213a8d772<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7685.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7685.dmg

File: libraries/shared/src/Debug.h (hunk 1 above)
Comment (zzmp) [inline libraries/shared/src/Debug.h]: Yes. I'm OK with that - I don't see why you would want to count negative, and seeing a crazy high number is a quick indicator that you've messed up.

**Comment by zzmp** [comment]
test this please

File: libraries/shared/src/Debug.h (hunk 1 above)
Comment (Atlante45) [inline libraries/shared/src/Debug.h]: Fair enough.
https://www.youtube.com/watch?v=R8sFE8_usN0

**Comment by hifi-gustavo** [comment]
Build console output for commit: cf9d160602e35c1dcc37ddf4b47cf8ba133b8905 available at: https://gist.github.com/d63f631183ddd1266082b27213a8d772<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7685.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7685.dmg

**Comment by zzmp** [comment]
thanks @hifi-gustavo!

**Comment by jherico** [comment]
The classes in the debug header all have generic names and should probably be in a `debug` namespace.

**Comment by zzmp** [comment]
@jherico something was already using the `debug` symbol so I took the path of least resistance, but I agree with you on namespacing. However, that's not the scope of this PR.

Generic names are on purpose, they're hopefully pretty generic things.

**Comment by kevinmthomas-carpool** [comment]
## QA Test
**PR Build:** 7685
**Testing Architecture:** 64-bit PC Desktop
**Testing OS:** Windows 10
**Testing Locations:** localhost, playa, cellscience, forum
**Smoke Test Results:** Partially successful test.

                                  |localhost |playa    |cellscience |forum  
----------------------------------|----------|---------|------------|-----
Render Rate:                      |59.11     |52.74    |57.86       |54.36    
Present Rate:                     |60.00     |60.02    |60.01       |60.00   
Simrate:                          |58.04     |52.13    |52.15       |53.00    
Render Stats Num Buffers:         |1201      |3218     |2721        |4472   
Render Stats gpu::Buffer Memory:  |28.8Mb    |64.8Mb   |31.2Mb      |57.4Mb        
Render Stats Num Textures:        |393       |529      |139         |920    
Render Stats gpu::Texture Memory: |1072.9Mb  |1081.7Mb |351.8Mb     |1854.4Mb         
Render Stats Triangles:           |685k      |782k     |480k        |1336k    
Render Stats Drawcalls:           |218       |447      |229         |770   
Render Stats Items:               |84        |258      |304         |406
GPU-Z Memory Size                 |4096Mb    |4096Mb   |4096Mb      |4096Mb
GPU-Z Initial Memory		  |225Mb     |225Mb    |225Mb       |225Mb
GPU-Z Memory Used		  |958Mb     |1156Mb   |799Mb       |1945Mb
GPU-Z Memory Load		  |8%        |9%       |7%          |10%

**localhost ‘createtestblocks.js’ Result:** Successful test.
**localhost ATP Upload (kelectricguitar10.fbx):** Failure.  Asset Browser - Asset Directory empty on ```localhost```, upon importing a model on ATP, process hangs, screen shot included.
**localhost ‘harmonicOscillator.js’ Result:** Successful test.
**localhost Echo Server Audio Result:** Partially successful test.  The ```Echo Local Audio``` was successful however when testing server audio on ```localhost```, it did not successfully echo back, upon visiting ```playa```, worked successfully upon revisiting ```localhost``` successful test.  Identical behavior in multiple prior PR's and have reported such.  Unsure if this is a real issue but wanted to bring it up for potential further analysis.  Enclosed is the log for further inspection.
**localhost ‘reverbTest.js’ Result:** Successful test.
**Additional Test Results Requested:** (none)
**Successful Logout/Crash:** Failure.  Upon visit of ```forum``` clicked on the tidy up button and it caused an instant crash.  Idential behavior in PR 7687, bugsplat successfully set as well as the log file included for deeper analysis.
[attachments.zip](https://github.com/highfidelity/hifi/files/222170/attachments.zip)

File: libraries/shared/src/Debug.h (hunk 1 above)
Comment (Atlante45) [inline libraries/shared/src/Debug.h]: @zzmp I see you haven't fixed that, were you serious?
I thought for sure you were joking.

**Pull request merged**